src/common/catch.tcl

```diff
@@ -0,0 +1,25 @@
+# Wrap the 'catch' command so that we can decide whether to catch signals
+# or propagate them as an error. Default is to propagate; to catch them use
+# 'catch -signal ...'
+if {[info commands builtin_catch] eq {}} {
+    rename catch builtin_catch
+
+    proc catch {args} {
+        set catch_signal no
+        if {[lindex $args 0] eq "-signal"} {
+            set catch_signal yes
+            set args [lrange $args 1 end]
+        }
+        set err [uplevel 1 [list builtin_catch {*}$args]]
+        if {$err == 1 && !$catch_signal} {
```

Comment:
Well if err can be a number other than 1 or 0 then it is not a boolean, so that probably won't work. And we certainly don't want to execute this code if err is something like TCL_BREAK.